pandas/tests/test_algos.py

```diff
@@ -767,7 +767,7 @@ def test_same_object_is_in(self):
         # with similar behavior, then we at least should
         # fall back to usual python's behavior: "a in [a] == True"
         class LikeNan:
-            def __eq__(self):
+            def __eq__(self, other: object):
```

Comment:
I _think_ we only need to correct the signature to resolve the error at this stage. @WillAyd 

```suggestion
            def __eq__(self, other):
```